### ISSUE TITLE
training code

### ISSUE THREAD

**Issue by MichaelSunEngineer** [Orange-OpenSource/AIVC#6]
Hi Dr. Theo,

Is it possible to share also the training code in the near future?
I really want to learn your training code.

Best regards

**Comment by MichaelSunEngineer** [comment]
> Hi Dr. Theo,
> 
> Is it possible to share also the training code in the near future? I really want to learn your training code.
> 
> Best regards

And I want to ask a question, is the FullNet in encode and decode the same model in the train process? Or did you train separately? How is the last loss constructed?

**Comment by theoladune** [comment]
Hi Michael, thanks for your interest in our work. Unfortunately, we don't plan to release training code in the near future due to our company policy.

The training process consists in learning the appropriate encoder and decoder operations, which are all gathered in FullNet. As such, they are optimized jointly to "mirror" each other i.e. the decoder must be able to undo what the decoder has done to compact the signal. The encoder and decoder are trained to optimize a rate-distortion loss in order to minimize visual degradations as well as the rate required to transmit the video.

**Comment by MichaelSunEngineer** [comment]
> Hi Michael, thanks for your interest in our work. Unfortunately, we don't plan to release training code in the near future due to our company policy.
> 
> The training process consists in learning the appropriate encoder and decoder operations, which are all gathered in FullNet. As such, they are optimized jointly to "mirror" each other i.e. the decoder must be able to undo what the decoder has done to compact the signal. The encoder and decoder are trained to optimize a rate-distortion loss in order to minimize visual degradations as well as the rate required to transmit the video.

Thank you very much for your reply. I have carefully read every line of your code and found some training clues. In fact, the training process is not divided into encode or decode，they are trained as a whole,  take some modifications when infer . The training process is included in the infer code. Am I right?

**Comment by theoladune** [comment]
Indeed, the training process is basically the inference + backpropagation to optimize the encoder and decoder weights together according to the loss value obtained during the inference. As such, training is built upon the forward function described in FullNet.
